lib/timers.js

```diff
@@ -477,47 +477,109 @@ exports.clearImmediate = function(immediate) {
 
 var unrefList, unrefTimer;
 
+function _makeTimerTimeout(timer) {
+  var domain = timer.domain;
+  var msecs = timer._idleTimeout;
+
+  // Timer has been unenrolled by another timer that fired at the same time,
+  // so don't make it timeout.
+  if (!msecs || msecs < 0)
+    return;
+
+  if (!timer._onTimeout)
+    return;
+
+  if (domain) {
+    if (domain._disposed)
+      return;
+
+    domain.enter();
+  }
+
+  debug('unreftimer firing timeout');
+  L.remove(timer);
+  timer._called = true;
+  _runOnTimeout(timer);
+
+  if (domain)
+    domain.exit();
+}
+
+function _runOnTimeout(timer) {
+  try {
+    var threw = true;
```

Comment:
I think @thefourtheye is asking for the stylistic change of moving the declaration out of the try block?